File: C++Project1/Grass.cpp
#include "Grass.h"

Grass::Grass(World* currentWorld, int posX, int posY, int age, bool newBorn)
    : Plant(currentWorld, 0, posX, posY, age, newBorn) {
};

void Grass::draw() {
    std::cout << char(177);
};

std::string Grass::getName() {
    return ("Grass");
};

Organism* Grass::clone(int posX, int posY) {
    return new Grass(currentWorld, posX, posY, 1, true);
};

Grass::~Grass() {
};

File: JavaProject2/Project2/Plants/Grass.java
package Project2.Plants;

import Project2.*;
import Project2.Extensions.*;
import java.awt.*;

public class Grass extends Plant {
    private static final int STRENGTH = 0;
    private static final int INITIATIVE = 0;

    public Grass(World world, Placement position, int age) {
        super(OrganismType.GRASS, world, position, age, STRENGTH, INITIATIVE);
        setColor(new Color(0, 204, 0));
    }

    @Override
    public String getName() {
        return "Grass";
    }
}
